app/services/developers/notification_service.rb

```diff
@@ -0,0 +1,16 @@
+module Developers
+  class NotificationService
+    def initialize(
+      webhook_service: WebhookDeliveries::CreateService.new,
+      event_publisher: Developers::EventPublisher.new
+    )
+      @webhook_service = webhook_service
+      @event_publisher = event_publisher
+    end
+
+    def execute(context, event_type, actor, resource)
```

Comment:
`context` should almost certainly be called `course`, since the downstream code does not accept any other value.

```suggestion
    def execute(course, event_type, actor, resource)
```

Reply:
I could imagine some scenarios where it will be triggered in context of school, not only course.